### PR TITLE
Bump Istio to 1.20

### PR DESCRIPTION
**Pull request by videlov**
**Description**

Changes proposed in this pull request:

- Bump Istio to 1.20
- https://istio.io/latest/news/releases/1.20.x/announcing-1.20/

**Related issue(s)**

https://github.com/kyma-project/istio/issues/477

### CODE REVIEW

**Review by barchw — CHANGES_REQUESTED** [review]

File: manifests/istio-operator-template.yaml [hunk 1]
```diff
@@ -276,9 +276,12 @@ spec:
         includeIPRanges: '*'
         logLevel: warning
         privileged: false
-        readinessFailureThreshold: 40
         readinessInitialDelaySeconds: 5
         readinessPeriodSeconds: 5
+        readinessFailureThreshold: 40
```
[the remark this answers is not in the document]
Comment (videlov) [reply manifests/istio-operator-template.yaml:281]: I did try them, but we introduce downtime with these settings. With our zero downtime test I was able to narrow it down to these new defaults. We will discuss it tomorrow after daily.

[the remark this answers is not in the document]
Comment (videlov) [reply manifests/istio-operator-template.yaml]: I did find an alternative in relaxing readiness settings for `istio-proxy`. Issue was with `readinessInitialDelaySeconds: 0` , I did leave it to 5 secs and can not observe downtime on Istio install or rollout restart of the deployment.

[the remark this answers is not in the document]
Comment (videlov) [reply manifests/istio-operator-template.yaml]: We decided to introduced the more relaxed polling period for readiness and keep the previous threshold of around 200 secs.

File: docs/release-notes/1.2.0.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+
+## New features
+
+- Update the Istio version to 1.20.0 [#483](https://github.com/kyma-project/istio/pull/483). Please check [Istio 1.20 Release Announcement](https://istio.io/latest/news/releases/1.20.x/announcing-1.20/) and [Upgrade Notes](https://istio.io/latest/news/releases/1.20.x/announcing-1.20/upgrade-notes/) for more details
+- Add Client IP in XFF header for AWS [#443](https://github.com/kyma-project/istio/pull/443)
```
Comment (nataliasitko) [inline docs/release-notes/1.2.0.md:5]:
```suggestion
- Add the client IP in the XFF header for AWS [#443](https://github.com/kyma-project/istio/pull/443)
```

**Review by barchw — APPROVED** [review]

**Review by nataliasitko — APPROVED** [review]

**Pull request merged**